sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py

```diff
@@ -333,11 +333,10 @@ def get_directory_properties(self, **kwargs):
             headers = kwargs.pop('headers', {})
             headers['x-ms-upn'] = str(upn)
             kwargs['headers'] = headers
-        return self._get_path_properties(cls=deserialize_dir_properties, **kwargs)
+        return cast(DirectoryProperties, self._get_path_properties(cls=deserialize_dir_properties, **kwargs))
```

Comment:
Note that there is a `cast` needed here because the call to `self._get_path_properties(cls=deserialize_dir_properties, **kwargs)` has these considerations:
1. The original `Union[DirectoryProperties, FIleProperties]` won't work
2. The new `BlobProperties` needs to be casted